### ISSUE TITLE
Switch component announces on and off states multiple times for NVDA users

### ISSUE THREAD

**Issue by LittleWhiteLoti** [indiana-university/rivet-source#682]
## Describe the bug
NVDA is reading the switch in a weird way
When off: On Off Off
When on: On Off On

## Expected behavior
It should only read the current state of on or off for the switch.

## Relevant context (required)
As there should only be one state active maybe it would be better to use a radio instead of a aria-checkbox and use a fieldset to wrap it. Additionally, since the switch is only meant to be visual, it should be fine to use checkboxes.

## Steps to reproduce
Use it in NVDA

**Comment by levimcg** [comment]
@LittleWhiteLoti I've passed the NVDA issue along to our Digital Accessibility Office for them to review, but we will need to know your **browser and OS version** so that we can confirm the bug and test any fixes.

FYI, I've updated the issue title to be more descriptive of the this specific to the potential bug you're reporting.

## The "switch" role

**mentioned in the original post:**

> As there should only be one state active maybe it would be better to use a radio instead of a aria-checkbox and use a fieldset to wrap it. Additionally, since the switch is only meant to be visual, it should be fine to use checkboxes.

This component has been reviewed by our Digital Accessibility Office and their recommendation is to use a button element along with the `aria-checked` and `role="switch"` attributes. This is also the [recommendation of the ARIA working group](https://w3c.github.io/aria/#switch) and the [MDN ARIA docs](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/switch_role).

The role of a _switch_ is different than the more generic checkbox (or radio) roles in that it is only meant to specifically convey an "on" or "off" state and must never have an _indeterminate_ state like a checkbox can have. The combination of the `switch` role and `aria-checked` attributes is what communicates this to assistive technologies. The use of the button element is intentional, and as far as any research and resources available are concerned, recommended.

Beyond semantics, the switch is not meant to participate in  `<form>` submission like a checkbox or radio buttons do. Instead it provides JavaScript hooks in the form of the `CustomEvent` interface which is good for doing asynchronous tasks (like an API call, etc.) that don't require it to be placed inside of a `<form>` element  and submitted like a checkbox or radio would.

**Comment by scottanthonymurray** [comment]
Fixed in `2.4.0`.
